agent.go

```diff
@@ -319,9 +321,25 @@ func (a *Agent) onSelectedCandidatePairChange(p *candidatePair) {
 
 func (a *Agent) startOnConnectionStateChangeRoutine() {
 	go func() {
-		for s := range a.chanState {
-			if hdlr, ok := a.onConnectionStateChangeHdlr.Load().(func(ConnectionState)); ok {
-				hdlr(s)
+		for {
+			select {
+			case c, isOpen := <-a.chanCandidate:
+				if !isOpen {
+					return
+				}
+
+				if onCandidateHdlr, ok := a.onCandidateHdlr.Load().(func(Candidate)); ok {
+					onCandidateHdlr(c)
+				}
+
+			case s, isOpen := <-a.chanState:
```

Comment:
After executing
```go
			close(agent.chanState)
			close(agent.chanCandidate)
```
order of these above cases are random, so I guess last `chanState` may be lost, if `case <-a.chanCandidate:` is called first.

I think it's better not to close `chanCandidate, chanState` and add `case <-a.done:` to this select. Then, we can close them in this routine.

Reply:
If I case on on `a.done` I lose the `ConnectionStateClosed`

`After calling close, and after any previously sent values have been received, receive operations will return the zero value for the channel's type without blocking.` It seems `value then closed` is getting me the behavior I want here.